src/data/set/list.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import data.set.image
+import data.list.basic
+import data.fin.basic
+
+/-!
+# Lemmas about `list`s and `set.range`
+
+In this file we prove lemmas about range of some operations on lists.
+-/
+
+open set
+
+namespace list
+
+variables {α β : Type*} (l : list α)
+
+lemma range_map (f : α → β) : set.range (map f) = {l | ∀ x ∈ l, x ∈ set.range f} :=
```

Comment:
I'm thinking, would this make more sense to name that `set.range_list_map`? This is not a lemma about `list.range`, so the current name is confusing (and I know you didn't pick it).